File: backend/src/NHLStats.Domain/Entities/Season.cs
using System.ComponentModel.DataAnnotations;
namespace NHLStats.Domain.Entities;

public class Season
{
    public int Id { get; set; }
    [Required]
    public string Name { get; set; } = null!;
    public int? HostedTeamId { get; set; }
    public DateTime StartedOn { get; set; }
    public SeasonStatus Status { get; set; } = SeasonStatus.Active;
    public int? ParentSeasonId { get; set; }
    public LeagueType LeagueType { get; set; } = LeagueType.NHL;

    public Team? HostedTeam { get; set; }
    public Season? ParentSeason { get; set; }
    public ICollection<SeasonUser> SeasonUsers { get; set; } = [];
    public ICollection<Match> Matches { get; set; } = [];
    public ICollection<RosterPlayer> RosterPlayers { get; set; } = [];
}

File: frontend/src/types/season.ts
import type { User } from './user'
import type { LeagueTypeValue } from './team'
import type { SeasonUserPositionCode } from './seasonUserPosition'
import type { GamingConsoleValue } from './gamingConsole'

export type SeasonStatus = 'Active' | 'Complete'

export interface Season {
    id: number
    name: string
    hostedTeamId: number | null
    hostedTeamName: string | null
    startedOn: string
    status: SeasonStatus
    parentSeasonId: number | null
    leagueType: LeagueTypeValue
    nhlYear: number | null
    console: GamingConsoleValue | null
}

export interface SeasonUser extends User {
    position: SeasonUserPositionCode | null
}

export interface SeasonDetail extends Season {
    users: SeasonUser[]
}

export interface CreateSeasonDto {
    name: string
    hostedTeamId?: number | null
    startedOn: string
    status?: string | null
    parentSeasonId?: number | null
    leagueType: LeagueTypeValue
    nhlYear?: number | null
    console?: GamingConsoleValue | null
}

export type UpdateSeasonDto = CreateSeasonDto

export interface ImportRealSeasonMatchesResult {
    imported: number
    skipped: number
    errors: string[]
}
